src/it/scala/org/broadinstitute/dsde/rawls/WorkspaceGenerator.scala

```diff
@@ -0,0 +1,183 @@
+package org.broadinstitute.dsde.rawls
+
+import org.broadinstitute.dsde.rawls.model._
+import org.broadinstitute.dsde.rawls.workspace.EntityUpdateOperations.{EntityUpdateOperation, RemoveAttribute, AddUpdateAttribute}
+import org.joda.time.DateTime
+
+import scala.util.Random
+
+import WorkspaceGenerator._
+
+object WorkspaceGenerator {
+  val rand = new Random
+
+  // TODO move these to config?
+  val objectNameLen = 10 // length of names for entities / method configs
+  val (minKeyLen, maxKeyLen) = (5, 20) // length of property key (for attribute, input, output, prereq, etc)
+  val (minStrLen, maxStrLen) = (10, 100) // length of value strings
+  val (minValSeqLen, maxValSeqLen) = (1, 100) // length of value lists
+
+  def randLen(min: Int, max: Int) = min + rand.nextInt(max - min)
+
+  def generateBoolean = AttributeBoolean(rand.nextBoolean())
+  def generateNumber = AttributeNumber(rand.nextDouble())
+
+  def generateString = AttributeString(rand.alphanumeric.take(randLen(minStrLen, maxStrLen)).mkString)
+  def generatePropertyKey = rand.alphanumeric.take(randLen(minKeyLen, maxKeyLen)).mkString
+  def generateObjectName = rand.alphanumeric.take(objectNameLen).mkString
+
+  def pickAttributeFrom(choices: String*) = AttributeString(choices(rand.nextInt(choices.length)))
+
+  def generateAnyPrimitive = rand.nextInt(3) match {
+    case 0 => generateBoolean
+    case 1 => generateNumber
+    case 2 => generateString
+  }
+
+  def generateValueList = {
+    val gen = for (i <- 0 to randLen(minValSeqLen, maxValSeqLen)) yield generateAnyPrimitive
+    AttributeValueList(gen.toSeq)
+  }
+
+  def generateAnyValue = rand.nextInt(5) match {
+    case 0 => generateValueList // 20% chance of list
+    case _ => generateAnyPrimitive
+  }
+
+  def generateEntityAnnotations(n: Int) = {
+    (for (i <- 0 to n) yield (generatePropertyKey, generateAnyValue)).toMap
+  }
+
+  def generateMethodConfigValue = rand.nextInt(3) match {
```

Comment:
```generateAnyPrimitive``` actually returns an ```AttributeValue``` that wraps a primitive, not the primitive itself - so calling ```toString``` would give you ```AttributeBoolean(true)``` instead of ```true```. It's a bit confusing since entities expect values to be wrapped in ```AttributeValue```, but method configs do not.